common/perception_utils/include/perception_utils/object_classification.hpp

```diff
@@ -23,72 +23,74 @@ namespace perception_utils
 {
 using autoware_auto_perception_msgs::msg::ObjectClassification;
 
-inline std::uint8_t getHighestProbLabel(
+inline ObjectClassification getHighestProbClassification(
   const std::vector<ObjectClassification> & object_classifications)
 {
-  std::uint8_t label = ObjectClassification::UNKNOWN;
-  float highest_prob = 0.0;
-  // TODO(Satoshi Tanaka): It might be simple if you use STL or range-v3.
-  for (const auto & object_classification : object_classifications) {
-    if (highest_prob < object_classification.probability) {
-      highest_prob = object_classification.probability;
-      label = object_classification.label;
-    }
+  if (object_classifications.empty()) {
+    return ObjectClassification{};
   }
-  return label;
+  return *std::max_element(
+    std::begin(object_classifications), std::end(object_classifications),
+    [](const auto & a, const auto & b) { return a.probability < b.probability; });
+}
+
+inline std::uint8_t getHighestProbLabel(
+  const std::vector<ObjectClassification> & object_classifications)
+{
+  auto classification = getHighestProbClassification(object_classifications);
+  return classification.label;
 }
 
-inline bool isVehicle(const uint8_t object_classification)
+inline bool isVehicle(const uint8_t label)
 {
-  return object_classification == ObjectClassification::BICYCLE ||
-         object_classification == ObjectClassification::BUS ||
-         object_classification == ObjectClassification::CAR ||
-         object_classification == ObjectClassification::MOTORCYCLE ||
-         object_classification == ObjectClassification::TRAILER ||
-         object_classification == ObjectClassification::TRUCK;
+  return label == ObjectClassification::BICYCLE || label == ObjectClassification::BUS ||
+         label == ObjectClassification::CAR || label == ObjectClassification::MOTORCYCLE ||
+         label == ObjectClassification::TRAILER || label == ObjectClassification::TRUCK;
+}
+
+inline bool isVehicle(const ObjectClassification & object_classification)
+{
+  return isVehicle(object_classification.label);
 }
 
 inline bool isVehicle(const std::vector<ObjectClassification> & object_classifications)
 {
   auto highest_prob_classification = getHighestProbLabel(object_classifications);
-  return highest_prob_classification == ObjectClassification::BICYCLE ||
-         highest_prob_classification == ObjectClassification::BUS ||
-         highest_prob_classification == ObjectClassification::CAR ||
-         highest_prob_classification == ObjectClassification::MOTORCYCLE ||
-         highest_prob_classification == ObjectClassification::TRAILER ||
-         highest_prob_classification == ObjectClassification::TRUCK;
+  return isVehicle(highest_prob_classification);
```

Comment:
`highest_prob_label` matches the output from the function.

Reply:
Thank you for polite review.
Fixed at [48a6966](https://github.com/autowarefoundation/autoware.universe/pull/2042/commits/48a69664b6693f05f2b449a3a23929a7e5d84922)